pkg/urbit/vere/pier.c

```diff
@@ -652,167 +596,136 @@ _pier_apply(u3_pier* pir_u)
   }
 }
 
-/* _pier_disk_load_precommit_file(): load precommit file into writ.
+/* _pier_set_ship():
 */
-static u3_writ*
-_pier_disk_load_precommit_file(u3_pier* pir_u,
-                               c3_d     lav_d,
-                               c3_c*    nam_c)
+static void
+_pier_set_ship(u3_pier* pir_u, u3_noun who, u3_noun fak)
 {
-  u3_writ* wit_u;
-  c3_d     evt_d;
-  c3_l     mug_l;
-  c3_d     pos_d;
-  c3_d*    buf_d;
-  c3_d     len_d;
-
-  if ( 2 != sscanf(nam_c, "%" PRIu64 "-%x.urbit-log", &evt_d, &mug_l) ) {
-    //  fprintf(stderr, "pier: load: precommit: bad file: %s\r\n", nam_c);
-    return 0;
-  }
+  c3_assert( c3y == u3ud(who) );
+  c3_assert( (c3y == fak) || (c3n == fak) );
 
-  wit_u = c3_calloc(sizeof(*wit_u));
+  c3_o fak_o = fak;
+  c3_d who_d[2];
 
-  //  fprintf(stderr, "pier: (%" PRIu64 "): %p restore\r\n", evt_d, wit_u);
+  u3r_chubs(0, 2, who_d, who);
 
-  wit_u->pir_u = pir_u;
-  wit_u->evt_d = evt_d;
+  c3_assert( ( (0 == pir_u->fak_o) &&
```

Comment:
Shouldn't this be `c3y`?

Reply:
That assertion is something like "these values in the pier are either uninitialized, or equal to the values in this function". I agree that that line looks odd, but maintain that it shouldn't be `c3y`. But maybe those values should be in a separately allocated structure, so I can just check for a null pointer.